internal/test/compilecheck/compile_check_test.go

```diff
@@ -0,0 +1,149 @@
+// requires a version of klauspost/compress that is not compatible with the Go
+// Driver. Must use GOWORK=off to run this test.
+
+const minSupportedVersion = "1.18"
+
+func TestCompileCheck(t *testing.T) {
+	cwd, err := os.Getwd()
+	require.NoError(t, err)
+
+	rootDir := filepath.Dir(filepath.Dir(filepath.Dir(cwd)))
+
+	versions, err := getDockerGolangImages()
+	require.NoError(t, err)
+
+	for _, version := range versions {
+		version := version // Capture range variable.
+
+		image := fmt.Sprintf("golang:%s", version)
+		t.Run(image, func(t *testing.T) {
+			t.Parallel()
+
+			req := testcontainers.ContainerRequest{
+				Image: image,
+				Cmd:   []string{"tail", "-f", "/dev/null"},
+				Mounts: []testcontainers.ContainerMount{
+					testcontainers.BindMount(rootDir, "/workspace"),
+				},
+				WorkingDir: "/workspace",
+				Env: map[string]string{
+					"GC": "go",
+					// Compilation modules are not part of the workspace as testcontainers requires
+					// a version of klauspost/compress not supported by the Go Driver / other modules
+					// in the workspace.
+					"GOWORK": "off",
+				},
+			}
+
+			genReq := testcontainers.GenericContainerRequest{
+				ContainerRequest: req,
+				Started:          true,
+			}
+
+			container, err := testcontainers.GenericContainer(context.Background(), genReq)
+			require.NoError(t, err)
+
+			defer func() {
+				err := container.Terminate(context.Background())
+				require.NoError(t, err)
+			}()
+
+			exitCode, outputReader, err := container.Exec(context.Background(), []string{"bash", "etc/compile_check.sh"})
+			require.NoError(t, err)
+
+			output, err := io.ReadAll(outputReader)
+			require.NoError(t, err)
+
+			t.Logf("output: %s", output)
+			assert.Equal(t, 0, exitCode)
+		})
+	}
+}
+
+// getDockerGolangImages retrieves the available Golang Docker image tags from
+// Docker Hub that are considered valid and meet the specified version
+// condition. It returns a slice of version strings in the MajorMinor format and
+// an error, if any.
+func getDockerGolangImages() ([]string, error) {
+	// URL to fetch the Golang tags from Docker Hub with a page size of 100
+	// records.
+	var url = "https://hub.docker.com/v2/repositories/library/golang/tags?page_size=100"
+
+	versionSet := map[string]bool{}
+	versions := []string{}
+
+	// Iteratively fetch and process tags from Docker Hub as long as there is a
+	// valid next page URL.
+	for url != "" {
+		resp, err := http.Get(url)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get response from Docker Hub: %w", err)
+		}
+
+		var data struct {
+			Results []struct {
+				Name string `json:"name"`
+			} `json:"results"`
+			Next string `json:"next"` // URL of the next page for pagination.
+		}
+
+		if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+			return nil, fmt.Errorf("failed to decode response Body from Docker Hub: %w", err)
+		}
+
+		resp.Body.Close()
```

Comment:
It's better to move the `Body.Close()` immediately after the error check of `http.Get()` and wrap it with a defer.

Reply:
Right... However, there is a risk that the current implementation may fail during decoding, leaving the body unclosed. Is it possible to wrap the HTTP Get in a separate function?

Reply:
I can update the code to close the response body if there is an error.

Reply:
It is somewhat hacky, but I think it's acceptable in a test...